src/llmtuner/tuner/core/loader.py

```diff
@@ -215,7 +215,10 @@ def load_model_and_tokenizer(
     # Prepare model for inference
     if not is_trainable:
         model.requires_grad_(False) # fix all model params
-        infer_dtype = torch.bfloat16 if torch.cuda.is_bf16_supported() else torch.float16 # detect cuda capability
+        if is_torch_npu_available():
+            infer_dtype = torch.float16
+        else:
+            infer_dtype = torch.bfloat16 if torch.cuda.is_bf16_supported() else torch.float16 # detect cuda capability
```

Comment:
Here is an if statement to evaluate the compatibility of the bf16 data type of the current computing environment.

Reply:
`torch.cuda.is_bf16_supported` will cause a runtime error on an NPU environment. 
FYI, to use PyTorch on NPU, users need to install both torch-cpu and [torch-npu](https://gitee.com/ascend/pytorch). As a result, cuda-related code is not included.